packages/picasso/src/LexicalEditorMaxIndentLevelPlugin/LexicalEditorMaxIndentLevelPlugin.ts

```
@@ -0,0 +1,81 @@
+import type { RangeSelection, ElementNode } from 'lexical'
+import { $getListDepth, $isListItemNode, $isListNode } from '@lexical/list'
+import { useLexicalComposerContext } from '@lexical/react/LexicalComposerContext'
+import {
+  $getSelection,
+  $isElementNode,
+  $isRangeSelection,
+  COMMAND_PRIORITY_CRITICAL,
+  INDENT_CONTENT_COMMAND,
+} from 'lexical'
+import { useEffect } from 'react'
+
+type Props = {
+  maxDepth?: number
+}
+
+const getElementNodesInSelection = (
+  selection: RangeSelection
+): Set<ElementNode> => {
+  const nodesInSelection = selection.getNodes()
+
+  if (nodesInSelection.length === 0) {
+    return new Set([
+      selection.anchor.getNode().getParentOrThrow(),
+      selection.focus.getNode().getParentOrThrow(),
+    ])
+  }
+
+  return new Set(
+    nodesInSelection.map(node =>
+      $isElementNode(node) ? node : node.getParentOrThrow()
+    )
+  )
+}
+
+const isIndentPermitted = (maxDepth: number): boolean => {
+  const selection = $getSelection()
+
+  if (!$isRangeSelection(selection)) {
+    return false
+  }
+
+  const elementNodesInSelection: Set<ElementNode> =
+    getElementNodesInSelection(selection)
+
+  let totalDepth = 0
+
+  for (const elementNode of elementNodesInSelection) {
+    if ($isListNode(elementNode)) {
+      totalDepth = Math.max($getListDepth(elementNode) + 1, totalDepth)
+    } else if ($isListItemNode(elementNode)) {
+      const parent = elementNode.getParent()
+
+      if (!$isListNode(parent)) {
+        throw new Error(
+          'ListMaxIndentLevelPlugin: A ListItemNode must have a ListNode for a parent.'
+        )
```

do you think we should raise an error here? I would just write in console the error and return false. wdyt?